hw02-arraylist/src/main/java/com/remifo/DIYArrayList.java

```diff
@@ -0,0 +1,280 @@
+package com.remifo;
+
+import java.util.*;
+
+public class DIYArrayList<T> implements List<T> {
+    private Object[] array = new Object[0];
+    private int size = 0;
+    private final int DEFAULT_CAPACITY = 10;
+
+    public DIYArrayList() { }
+
+    @Override
+    public int size() {
+        return this.size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.size == 0;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return indexOf(0) >= 0;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new Itr();
+    }
+
+    @Override
+    public Object[] toArray() {
+        return Arrays.copyOf(this.array, this.size);
+    }
+
+    @Override
+    public <T1> T1[] toArray(T1[] a) {
+        a = (T1[]) Arrays.copyOf(this.array, this.size);
+        return a;
+    }
+
+    @Override
+    public boolean add(T t) {
+        add(this.size, t);
+        return true;
+    }
+
+    @Override
+    public void add(int index, T element) {
+        if (index >= 0 && index <= this.size) {
+            if (this.array.length >= this.size)
+                grow(++this.size);
```

Comment:
Лучше разбивать такое на две строки. 
Не все хорошо понимают такие записи, что чревато ошибками.

```suggestion
                this.size++;
                grow(this.size);
```

В других подобных местах аналогично.

Reply:
Хорошо